### PR TITLE
Group api

### PR DESCRIPTION
**Pull request by thiagogquinto**
### Contributor checklist

- [X] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

- Added the HTTP methods for the Group model
- The validations for the Group were removed
- The model OrganizationApplicationStatus was removed

### Related issue

- #671

### CODE REVIEW

File: backend/entities/views.py [hunk 1]
```diff
@@ -1,250 +1,250 @@
...
+        
+        if request.user != group.created_by:
+            return Response({"error": "You are not authorized to update this group"}, status.HTTP_401_UNAUTHORIZED)
+        
+        serializer = self.get_serializer(group, data=request.data, partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+    
+    def destroy(self, request: Request, *args: str, **kwargs: int) -> Response:
```
Comment (to-sta) [inline backend/entities/views.py:199]: @thiagogquinto 
When we are deleting the group, we want to make sure that only the creator is allowed to do that. 

@andrewtavis 
At the moment, when we delete the group, we set the deletion_date. But there should be more to this, like:

1. Delete the group and remove the `deletion_date` field
2. Set fields to blank/NULL on deletion

**Review by to-sta — CHANGES_REQUESTED** [review]

File: backend/entities/views.py (hunk 1 above)
Comment (andrewtavis) [reply backend/entities/views.py:199]: I don't think we need to worry as much about keeping minor data for groups as we do for for orgs. I think the first option would be fine? Saves on storage as well.

Comment (andrewtavis) [reply backend/entities/views.py:199]: But would you suggest either, @to-sta?

**Comment by andrewtavis** [comment]
Thanks for the review, @to-sta! @thiagogquinto, let us know if you need any help with the refactoring or if there are any questions!

File: backend/entities/views.py (hunk 1 above)
Comment (to-sta) [reply backend/entities/views.py:199]: I guess it's not necessary to set values to blank/NULL. I would suggest to delete the entry and remove the deletion_date field.

**Review by to-sta — APPROVED** [review]

**Review by andrewtavis — APPROVED** [review]
Thanks for the fixes here, @to-sta, and for the continued work, @thiagogquinto! Hope to see you at the dev sync on Saturday! 😊

**Pull request merged**